### ISSUE TITLE
VO stops automatic read through

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/ratio-and-proportion#347]
For https://github.com/phetsims/QA/issues/601.
MacOS 11.2, Safari with VO.
From Slack:
>On RaP with Mac VO, I noticed that the screen reader would stop once it reached the part where both hands could move, even if reading automatically. It was fairly easy to move past it though. Is this a bug?
Hmmm, perhaps that is because it is a custom div with application role? We don't have any other occurrences in sims where the application role (custom interaction) is available in the generic nav order (grab/drag interaction hides it behind the grab button). Perhaps that is the reasoning. We would like want to double check with @terracoda to be sure about it.
Our resident genius JG figured out that it is because we are doing an aria-live alert upon focusing the both-hands interaction. @KatieWoe please assign @terracoda to determine what the best design solution for this issue is. I'll try to encapsulate the problem in the issue for her. (edited) 

https://user-images.githubusercontent.com/41024075/107101262-fd474980-67d3-11eb-92ba-b3aec1b2da0f.MOV

**Comment by zepumph** [comment]
@jessegreenberg mentioned that this is because we alert on focus of the both-hands interaction. That alert interrupts the read-through. @terracoda please see if you can reproduce, and recommend how best to proceed. We may decide that it isn't worth solving, because it is too valuable to have an alert on focus. Perhaps also we could experiment with making that alert polite on focus, but assertive in other cases.

**Comment by terracoda** [comment]
_Note to self - this is not an issue on iOS as we do not have the Both Hands interaction available on iOS VO._

I have noticed this issue in the past, and it is a little inconsistent. I am not sure if the inconsistency has to do with OS version or browser caching. Just now while testing, here's what I experienced on Create Screen:

1. I selected Create Screen from the Home screen, auto reading read straight through to the Reset all button:
![Screen Shot 2021-02-08 at 10 32 02 AM](https://user-images.githubusercontent.com/1902116/107230764-07e32800-69fa-11eb-8006-fdf53a6e7432.png)

Same thing happened on Discover Screen
![Screen Shot 2021-02-08 at 10 47 26 AM](https://user-images.githubusercontent.com/1902116/107231677-2695ee80-69fb-11eb-99da-765fbf2a2269.png)
I think the entire PDOM is presented as a description (maybe??) and I don't hear all the role information. I don't think I hear slider, and button, for example. Though I hear group. 


2. I noticed that the auto-read on page load behaves differently from the auto-read with the VO command, "read from beginning (Ctrl+Option+B)". With the VO command, the first time through it read all the way to the PhET Menu. I don't think that I have seen this before with this sim. AND then on the second and third read-from-beginning comands, it stopped at the Both Hands interaction. When reading with the VO command, I hear all document and interaction semantics.

I am happy to know why this happens, i.e., aria-live alert on focus interrupts read through.

From a design perspective, I think the auto-reading of the scene summary is most critical, and as @KatieWoe mentioned, it is easy to continue on by using the arrow keys. I don't think the stopping on auto-read is super critical.

I think what is more problematic is that the aria-live alert (the Both Hands current value object response), "hands, at challenge ratio" interrupts the focus event utterance of the Both Hands interaction which is, "Both Hands, movable".

I have been concerned about this interruption, but never made an issue, as we had bigger fish to fry and, and I figured it was highly dependent on screen reader voice rate. I just tested with a slower voice rate, and the issue is more pronounced the slower the speech rate.

I decided to review what happens with other custom interactions (BASE's Yellow Balloon and Friction's Chemistry Book). Because these 2 interactions are two-step interactions, we can provide a very customized string on grab, and this response can be different depending on whether it is the initial grab (adding extra instructions) or not (just providing relevant position information).

In the case of RaP, it is not a two-step grab interaction, so we are trying to deliver two sequential responses that provide current value (proximity to ratio) and the objects name and role (Both Hands, movable).

Personally, for the on focus response, I feel hearing "Both Hands, movable" is more important than hearing the current proximity to ratio.

Since this is our first already grabbed movable object, I am not a 100 % sure of the best approach. I think we have design 3 options:

1. Do nothing. Keep the response as-is, and do not not change the timing, and continuing working on this issue with the Self-Voicing design.
2. Remove the initial object value response when we have a custom object like this that cannot avail of aria-valuetext. AND to compensate ensure a complete dynamic state description is nearby. We are already partially doing this with the first sentence in the Play Area for RaP (distance between hands + position of each hand is described). The scene summary has the full current value description (proximity to ratio, distance between hands, and position of each hand), it is just a little farther away. 
3. Keep the response (and design pattern), and adjust adjust timing of object value response on a sim by sim basis, so it is less likely to obscure the interaction's name and role. This might be tricky for slow speech rates, regardless of any timing change, AND the current timing might not be an issue for most screen reader users, as many use a faster speech rate than me.

@zepumph, I have a feeling that option 2 might be the best long-term solution, but I feel like option 1 is reasonable path forward given reasons on option 3 (less of an issue the faster speech rates).

**Comment by terracoda** [comment]
I removed the bug label as I think this a design problem not a bug.

**Comment by zepumph** [comment]
Thanks for your helpful summary. I wanted to pose two more potential options, though likely option (1) is what we should do.

(4): I was curious if this was a problem mainly because the focus alert was "assertive" instead of "polite", above I committed a change to master where the object response that occurs from changing the values with both hands is still assertive, but the focus version of this description is polite. @terracoda can you please see if the behavior is more reasonable when the alert is polite instead of assertive during this on focus description alert?

(5): This is basically just 2 "prime", but what about just removing this content on focusable, and not updating description around it. Then you only get object description when you interactive with it. If you want to know about its current state, you can go to the play area. The content is still there, just a bit farther away from the interaction.


I don't know that (3) is really viable, because it seems too dependent on the screen reader settings whether or not the alert interrupted. I don't think this is something we can control with timing. 

@terracoda please take a look if you can today, and I will be excited to discuss tomorrow further with you.

**Comment by terracoda** [comment]
@zepumph, option (4) is a bull's eye! The polite setting for the object response works great!
On focus I hear:
- Both Hands, movable 
- hands, at challenge ratio

No interrupting on "Both Hands, movable"!

And I don't think it is a bug that the VO stops reading at the Both Hands interaction.

I think we can close this issue, if you are ready to close it.

**Comment by terracoda** [comment]
Re-open if you need to.

**Comment by zepumph** [comment]
I will need to cherry pick into the branch. Glad that worked out!

**Comment by samreid** [comment]
Reopening based on a TODO in the code.  I'm not sure if the TODO can be deleted or if it needs attention.
